catroid/src/main/java/org/catrobat/catroid/content/bricks/UserDefinedBrick.java

```diff
@@ -55,144 +55,175 @@
 		if (lastContentIsLabel()) {
-			userBrickDataList.remove(userBrickDataList.size() - 1);
+			userDefinedBrickDataList.remove(userDefinedBrickDataList.size() - 1);
 		}
 	}
 
-	public void addInput(Nameable input) {
-		userBrickDataList.add(new UserBrickInput(input));
+	public void addInput(String input) {
+		userDefinedBrickDataList.add(new UserDefinedBrickInput(input));
 	}
 
 	public boolean isEmpty() {
-		return userBrickDataList.isEmpty();
+		return userDefinedBrickDataList.isEmpty();
+	}
+
+	public List<UserDefinedBrickData> getUserDefinedBrickDataList() {
+		return this.userDefinedBrickDataList;
 	}
 
-	public List<Nameable> getUserDataList(boolean inputOrLabel) {
-		List<Nameable> userDataList = new ArrayList<>();
-		if (inputOrLabel) {
-			for (UserBrickData userBrickData : userBrickDataList) {
-				if (userBrickData instanceof UserBrickInput) {
-					userDataList.add(((UserBrickInput) userBrickData).getInput());
+	public List<String> getUserDataList(boolean isInput) {
+		List<String> userDataList = new ArrayList<>();
+		if (isInput) {
+			for (UserDefinedBrickData userDefinedBrickData : userDefinedBrickDataList) {
+				if (userDefinedBrickData instanceof UserDefinedBrickInput) {
+					userDataList.add(((UserDefinedBrickInput) userDefinedBrickData).getInput());
 				}
 			}
 		} else {
-			for (UserBrickData userBrickData : userBrickDataList) {
-				if (userBrickData instanceof UserBrickLabel) {
-					userDataList.add(((UserBrickLabel) userBrickData).getLabel());
+			for (UserDefinedBrickData userDefinedBrickData : userDefinedBrickDataList) {
+				if (userDefinedBrickData instanceof UserDefinedBrickLabel) {
+					userDataList.add(((UserDefinedBrickLabel) userDefinedBrickData).getLabel());
 				}
 			}
 		}
 		return userDataList;
 	}
 
 	private boolean lastContentIsLabel() {
-		if (userBrickDataList.isEmpty()) {
+		if (userDefinedBrickDataList.isEmpty()) {
 			return false;
 		}
-		return userBrickDataList.get(userBrickDataList.size() - 1) instanceof UserBrickLabel;
+		return userDefinedBrickDataList.get(userDefinedBrickDataList.size() - 1) instanceof UserDefinedBrickLabel;
+	}
+
+	public boolean isUserDefinedBrickDataEqual(UserDefinedBrick other) {
+		if (userDefinedBrickDataList.size() != other.userDefinedBrickDataList.size()) {
+			return false;
+		}
+		for (int dataIndex = 0; dataIndex < userDefinedBrickDataList.size(); dataIndex++) {
+			UserDefinedBrickData thisData = userDefinedBrickDataList.get(dataIndex);
+			UserDefinedBrickData otherData = other.userDefinedBrickDataList.get(dataIndex);
+
+			if (!thisData.getClass().equals(otherData.getClass())) {
+				return false;
+			}
+			if (thisData instanceof UserDefinedBrickLabel && !thisData.equals(otherData)) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	public Brick clone() throws CloneNotSupportedException {
+		UserDefinedBrick clone = (UserDefinedBrick) super.clone();
+		clone.userDefinedBrickDataList = this.getUserDefinedBrickDataList();
```

Comment:
The List of the clone will point to the list of the original object. If only the clones are present anymore, the list wont